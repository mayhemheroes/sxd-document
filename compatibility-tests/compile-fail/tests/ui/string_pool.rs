use sxd_document::__internal::StringPool;

fn string_cannot_outlive_the_pool() {
    let _s = {
        let pool = StringPool::new();
        pool.intern("hello")
    };
}

fn main() {}
