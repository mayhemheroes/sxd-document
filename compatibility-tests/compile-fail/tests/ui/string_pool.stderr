error[E0597]: `pool` does not live long enough
 --> tests/ui/string_pool.rs:6:9
  |
4 |     let _s = {
  |         -- borrow later stored here
5 |         let pool = StringPool::new();
  |             ---- binding `pool` declared here
6 |         pool.intern("hello")
  |         ^^^^ borrowed value does not live long enough
7 |     };
  |     - `pool` dropped here while still borrowed
